### PR TITLE
Add ability to Share image from Status

### LINKED ISSUE #12472: Add ability to Share image from Status
# Feature Issue

[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to Share an image **so that** I can send images to apps and contacts outside of Status. 
As a user I want to Save an image **so that** I can keep it for later use.

### Description

[comment]: # (Describe the feature you would like and include designs and Figma links if available.)
[comment]: # (Describe the use cases the implementation should cover. E.g. New users, existing users, public chat, group chat, etc)
*Summary*: As per feature request #11912 

Share image comes with a [design update](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=4190%3A24449) that should apply to Android and iOS. Currently the implementation differs between Android and iOS (see #12474) 

### Acceptance Criteria

- Share option is available when viewing image in full screen
- UI is update to reflect design update

[comment]: # (Rules for the future PR to be accepted.)

### Notes

- Note that this request is somewhat orthogonal to e.g. blocking screenshots and might need community debate to understand if easily sharing images outside of Status is desirable from a privacy point of view. That said, the ability to share is already available albeit with a workaround by either saving and sharing the image or enabling screenshots and sharing an image 
- Note that this issue is distinct from #12148, which describes ability to Share an image to Status

[comment]: # (For the developer who will implement. Include planning poker estimates, guidance, etc.)

### PR DESCRIPTION
**Pull request by briansztamfater**
fixes #12472 

### Summary

This PR adds the ability to share an image to other apps from the image preview in Status.

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

### Steps to test

- Open Status
- Send image from user A to user B
- Open image preview
- Verify share functionality and UI changes

status: ready

### CODE REVIEW

File: src/status_im/utils/share.cljs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+(ns status-im.utils.share
+  (:require ["react-native-share" :default react-native-share]))
+
+(defn open [options on-success on-error]
+  (println react-native-share)
```
Comment (flexsurfer) [inline src/status_im/utils/share.cljs]: leftover

**Comment by flexsurfer** [comment]
just wondering why two icons ?

**Comment by shivekkhurana** [comment]
This is a timely PR. We need this feature for sharing NFTs too !!

**Comment by briansztamfater** [comment]
> just wondering why two icons ?

One is for download the image (that was already implemented, but the UI was changed) and the other is for sharing it.

File: src/status_im/utils/share.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im/utils/share.cljs]: Removed

**Comment by flexsurfer** [comment]
> > just wondering why two icons ?
> 
> One is for download the image (that was already implemented, but the UI was changed) and the other is for sharing it.

but we now have save option in share ,so we don't need download icon anymore ?

**Review by flexsurfer — APPROVED** [review]

**Comment by briansztamfater** [comment]
> > > just wondering why two icons ?
> > 
> > 
> > One is for download the image (that was already implemented, but the UI was changed) and the other is for sharing it.
> 
> but we now have save option in share ,so we don't need download icon anymore ?

Yeah it is valid concern, @hesterbruikman any thoughts on this?

**Comment by qoqobolo** [comment]
@briansztamfater thanks for adding this! Works well except for these two issues.

### ISSUE 1: Mismatch with design for vertical / zoomed images

Screenshot 2021-09-06 at 12 47 10

[Figma](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=9994%3A11):

Screenshot 2021-09-06 at 12 48 04

### ISSUE 2: Can't share an image to Telegram / Mail on iOS

Get endless spinner when trying to share an image to Telegram and don't receive an email with shared image when sharing to Mail.
However, everything is fine with other apps, e.g. Facebook, Instagram, etc, so not sure, that this is a Status app issue.
Could you take a look, please?

Screenshot 2021-09-06 at 12 52 54

**Comment by briansztamfater** [comment]
Thanks @qoqobolo, I'll take a look at those issues and come back to you

**Comment by briansztamfater** [comment]
@qoqobolo Both issues should be fixed now!

**Comment by churik** [comment]
Can you please add accessibility-ids for new elements?
Required for e2e.
Thanks.
@briansztamfater

**Comment by briansztamfater** [comment]
> Can you please add accessibility-ids for new elements?
> Required for e2e.
> Thanks.
> @briansztamfater

@qoqobolo Added accessibility-ids to all close, save and share buttons. Let me know if that's ok and I can merge then :)
Thanks!

**Comment by qoqobolo** [comment]
@briansztamfater thanks a lot! 
Yep, PR is ready to be merged :)

**Pull request merged**